src/communication/connection.js

```diff
@@ -0,0 +1,11 @@
+
+define(["communication/session"], function (Session) {
+    "use strict";
+
+    var Connection = function (serverUrl, callback, callbackError, publisher) {
```

Comment:
You should have only 1 callback (instead of 2)
try to follow nodejs rules (ex: function(error, data){...}
choose a better naming connectionListener for instance